lib/SPIRV/Lowering.md

```diff
@@ -0,0 +1,132 @@
+The SPIRV-LLVM-Translator will "lower" some LLVM intrinsic calls to another function or implementation
+in one of three ways:
+
+1. In transIntrinsicInst in SPIRVWriter.cpp, calls to LLVM intrinsics are replaced with a SPIRV ExtInst.
+   For example:
+
+        %0 = tail call i32 @llvm.ctlz.i32(i32 %x, i1 true)
+
+   is translated into SPIRV with an SPIRV ExtInst clz:
+
+        6 ExtInst 2 7 1 clz 5
+
+   The code in transIntrinsicInst to do this translation is:
+   
+          case Intrinsic::ctlz:
+          case Intrinsic::cttz: {
+            SPIRVWord ExtOp = IID == Intrinsic::ctlz ? OpenCLLIB::Clz : OpenCLLIB::Ctz;
+            SPIRVType *Ty = transType(II->getType());
+            std::vector<SPIRVValue *> Ops(1, transValue(II->getArgOperand(0), BB));
+            return BM->addExtInst(Ty, BM->getExtInstSetId(SPIRVEIS_OpenCL), ExtOp, Ops,
+                                  BB);
+          }
+
+   When these ExtInst are reverse translated they are converted to calls:
```

Comment:
Can you please document what happens when reverse translation is called with --spirv-target-env=SPV-IR i.e the user requests SPIR-V friendly IR?

Reply:
I've added a little documentation, but this is just duplicating text from the --help option.   I do not understand enough to explain why --spirv-target-env=SPV-IR should be used.  Perhaps someone else can re-write this seciton.